tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -490,17 +491,20 @@ def flatten(yml):
     return flatten(yml)
 
 
-def filter_gitlab_ci_configuration(yml: dict, job: str | None = None) -> dict:
+def filter_gitlab_ci_configuration(yml: dict, job: str | None = None, keep_special_objects: bool = False) -> dict:
     """
     Filters gitlab-ci configuration jobs
 
     - job: If provided, retrieve only this job
+    - keep_special_objects: Will keep special objects (not jobs) in the configuration (variables, stages, etc.)
```

Comment:
why do we need to dismiss them?

Reply:
Some tasks process only jobs (e.g. the linter for `changes: paths: ...`) so these objects should not be always added